src/components/mgt-person-card/mgt-person-card.ts

```diff
@@ -498,16 +504,17 @@ export class MgtPersonCard extends MgtTemplatedComponent {
   }
 
   /**
-   * Use the sip: protocol to initiate a chat message to the user.
+   * Initiate a chat message to the user.
    *
    * @protected
    * @memberof MgtPersonCard
    */
   protected chatUser() {
     const user = this.personDetails as MicrosoftGraph.User;
-    if (user && user.mailNickname) {
-      const chat = user.mailNickname;
-      window.open('sip:' + chat, '_blank');
+    if (user && user.userPrincipalName) {
+      const users: string = [user.userPrincipalName].join(',');
+      const url = `https://teams.microsoft.com/l/chat/0/0?users=${users}`;
+      window.open(url, '_blank');
```

Comment:
This approach still opens a new window when the app is running as a Teams tab. Could we do a check here if we are running in Teams and use the [`executeDeepLink`](https://docs.microsoft.com/en-us/javascript/api/@microsoft/teams-js/microsoftteams?view=msteams-client-js-latest#executedeeplink-string---status--boolean--reason---string-----void-) function? What if we do something similar to what we do in the [TeamsProvider](https://github.com/microsoftgraph/microsoft-graph-toolkit/blob/master/src/providers/TeamsProvider.ts#L131)? We could probably factor out the `microsoftTeamsLib` to a static helper that can also be used here.

Reply:
Excellent call out. I'll add this